main.go

```diff
@@ -148,3 +163,19 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// parseLogLevel converts a string log level to zapcore.Level
+func parseLogLevel(level string) (zapcore.Level, bool) {
+	switch strings.ToLower(level) {
+	case "debug":
+		return zapcore.DebugLevel, true
+	case "info":
+		return zapcore.InfoLevel, true
+	case "warn", "warning":
+		return zapcore.WarnLevel, true
+	case "error":
+		return zapcore.ErrorLevel, true
+	default:
+		return zapcore.InfoLevel, false // default to info if invalid level provided
+	}
+}
```

Comment:
I think `if err != nil` is a common practice. What is the reason of returning a bool instead of an error directly?

Reply:
Discussed